### ISSUE TITLE
Update install instructions for debian-testing

### ISSUE THREAD

**Issue by DentonGentry** [tailscale/tailscale#1876]
**Describe the bug**
apt-key in Debian-testing is deprecated and is intended to be disabled in Debian 11 and Ubuntu 22.04
https://www.linuxuprising.com/2021/01/apt-key-is-deprecated-how-to-add.html

**To Reproduce**
`curl -fsSL https://pkgs.tailscale.com/stable/debian/bullseye.gpg | sudo apt-key add -`
generates an error that apt-key is deprecated.

**Expected behavior**
updated instructions

**Version information:**
 - Device: PC
 - OS: Linux
 - OS version: Debian Testing
 - Tailscale version: N/A

**Comment by DentonGentry** [comment]
I believe we change [the instructions](https://tailscale.com/kb/1038/install-debian-bullseye/) from this:
`curl -fsSL https://pkgs.tailscale.com/stable/debian/bullseye.gpg | sudo apt-key add -`
to this:
`curl -fsSL https://pkgs.tailscale.com/stable/debian/bullseye.gpg | gpg --dearmor | sudo tee /usr/share/keyrings/tailscale-archive-keyring.gpg | gpg --armor`

and we need to update the contents of https://pkgs.tailscale.com/stable/debian/bullseye.list from:
```
# Tailscale packages for debian bullseye
deb https://pkgs.tailscale.com/stable/debian bullseye main
```
to:
```
# Tailscale packages for debian bullseye
deb [signed-by=/usr/share/keyrings/tailscale-archive-keyring.gpg] <https://pkgs.tailscale.com/stable/debian bullseye main>
```

**Comment by kpfleming** [comment]
Clarification: apt-key will *not* be disabled in Debian 11. It will be available, but deprecated, It will disappear in Debian 12. It's still a good idea to make these changes to avoid the deprecation warning messages being shown to users, but it's not urgent.

**Comment by adriel** [comment]
https://tailscale.com/download/linux/debian-bullseye
FYI, the instructions here for bullseye still say to use apt-key and result in errors when trying to run the apt-get update command.

```
Get:9 https://pkgs.tailscale.com/stable/debian bullseye InRelease                  
Err:9 https://pkgs.tailscale.com/stable/debian bullseye InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 458CA832957F5868
Reading package lists... Done
W: GPG error: https://pkgs.tailscale.com/stable/debian bullseye InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 458CA832957F5868
E: The repository 'https://pkgs.tailscale.com/stable/debian bullseye InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

**Comment by nikolay-turpitko** [comment]
> https://pkgs.tailscale.com/stable/#debian-bullseye However, the instructions here do work on Bullseye.

Neither worked for me. NO_PUBKEY error in all cases.
